include/evmc/filter_iterator.hpp

```diff
@@ -54,26 +64,32 @@ struct filter_iterator
     }
 
 public:
+    /// The constructor of the base iterator pair.
     constexpr filter_iterator(BaseIterator it, BaseIterator end) noexcept : base{it}, base_end{end}
     {
         forward_to_next_value();
     }
 
+    /// The dereference operator.
     constexpr auto operator*() noexcept
     {
         // We should not read from an input base iterator twice. So the only read is in
         // forward_to_next_value() and here we return the cached value.
         return value;
     }
 
+    /// The increment operator.
     constexpr void operator++() noexcept
     {
         ++base;
         forward_to_next_value();
     }
 
-    constexpr bool operator!=(const filter_iterator& o) const noexcept { return base != o.base; }
+    /// The equal operator.
```

Comment:
I think it should be "equality"/"inequality" operator 
https://docs.microsoft.com/en-us/cpp/cpp/equality-operators-equal-equal-and-exclpt-equal?view=msvc-170